sdk/core/azure_core_test/src/perf/mod.rs

```diff
@@ -535,6 +544,23 @@ impl PerfRunner {
     }
 }
 
+/// Instrument a client options appropriately for a perf test.
+///
+/// # Arguments:
+///
+/// - recording: Test recording.
+/// - client_options: Core client options
+///
+pub fn instrument_for_perf_test(recording: &Recording, client_options: &mut ClientOptions) {
```

Comment:
Do we need this as opposed to just passing in the `RecordingOptions`? Just wondering if it's worth increasing the API surface.

Reply:
I added this earlier today because I realized yesterday that the "RemoveRecording: false" experience felt unnatural (the header is a negative state and I felt it was important to preserve the header semantics rather than invert the option. 

Adding the "instrument for perf test" method hides this complexity and removes an opportunity for client teams to screw this up.